src/wallet/spend.cpp

```diff
@@ -162,6 +162,8 @@ void AvailableCoins(const CWallet& wallet, std::vector<COutput>& vCoins, const C
             continue;
         }
 
+        bool tx_from_me = CachedTxIsFromMe(wallet, wtx, ISMINE_ALL);
```

Comment:
In commit "wallet: Store whether a COutput is from the wallet" (521a703d056c13d85dbaabd289d2cc7edfd671c0)

This seems ok, but just reading commit description and looking at this code this doesn't seem to be an improvement. Two CachedTxIsFromMe calls are added and two are dropped which is neutral, but COutput parameter list is growing even more, which seems bad.

I'm assuming something justifies this in a later commit, but it would be good to hint at whatever this might be in commit description,

EDIT: After reviewing PR, still not exactly clear what benefit of this change is. But seems basically neutral.

Reply:
The purpose is that when filtering UTXOs, we need to know whether they belong to the wallet. Since we will be removing access to CWallet and CWalletTx in the future, this information must be stored in COutput itself.